plugins/destination/mssql/client/migrate.go

```diff
@@ -17,16 +17,21 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
 	return nil
 }
 
-func (c *Client) migrate(ctx context.Context, table *schema.Table) error {
+func (c *Client) migrate(ctx context.Context, table *schema.Table) (err error) {
 	c.logger.Info().Str("table", table.Name).Msg("Migrating table started")
 	defer c.logger.Info().Str("table", table.Name).Msg("Migrating table done")
+	defer func() {
+		if err != nil {
+			c.logErr(err)
+		}
+	}()
 
 	if len(table.Columns) == 0 {
 		c.logger.Info().Str("table", table.Name).Msg("Table with no columns, skipping")
 		return nil
 	}
 
-	switch tableExist, err := c.tableExists(ctx, table.Name); {
+	switch tableExist, err := c.tableExists(ctx, table); {
```

Comment:
I could be missing something, but it looks like `err` is being shadowed here (and inside the cases), so I don't think the `migrate`-scope `err` that is logged in the defer will ever be non-nil?

Reply:
The `err` in the defer is coming from the named return argument I think.

Reply:
Right, but where is it getting set?

Reply:
yes, the err in defer is exactly from the named ret value

Reply:
on return

Reply:
Oh okay, I didn't know defer worked that way with named return values, TIL

Reply:
yeah, basically, you can use defer to update named return values, see https://go.dev/blog/defer-panic-and-recover